src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,50 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask = 1L << 63;
 
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.X64.IsSupported)
+            {
+                // Create vectors of the elements, and make them float to allow using SSE rather than SSE2 instructions
+                var xvec = Vector128.CreateScalarUnsafe(x).AsSingle();
+                var yvec = Vector128.CreateScalarUnsafe(y).AsSingle();
 
-            // If the sign bits of x and y are not the same,
-            // flip the sign bit of x and return the new value;
-            // otherwise, just return x
+                // Remove the sign from x, and remove everything but the sign from y
+                // Creating from a 'const long' is better than from the correct 'const float/double'
+                var mask = Vector128.CreateScalarUnsafe(signMask).AsSingle();
+                yvec = Sse.And(yvec, mask);
+                xvec = Sse.AndNot(mask, xvec);
 
-            if ((xbits ^ ybits) < 0)
+                // Simply OR them to get the correct sign
+                return Sse.Or(xvec, yvec).AsDouble().ToScalar();
+            }
+            else
             {
-                return BitConverter.Int64BitsToDouble(xbits ^ long.MinValue);
+                return SoftwareFallback(x, y);
             }
 
-            return x;
+            static double SoftwareFallback(double x, double y)
+            {
+                const long signMask
```

Comment:
Nit: the two definitions of `const long signMask` are duplicated in different format.
Constants won't be treated as a capture of local function. It will be expanded at compile time instead. Just one definition outside the local function is enough.